services/core-api/app/api/services/ams_api_service.py

```diff
@@ -0,0 +1,169 @@
+        permit_type_mapping = {
+            "PER": "Permit",
+            "APP": "Approval"
+        }
+        return permit_type_mapping[data]
+
+    @classmethod
+    def __create_full_address(cls, address_line1, city, sub_division_code, post_code):
+        return f"{address_line1}, {city}, {sub_division_code}, {post_code}"
+
+    @classmethod
+    def __get_new_authorization_details(cls, authorizations_data):
+        new_authorization_values = []
+
+        if len(authorizations_data['new']) > 0:
+            for item in authorizations_data['new']:
+                new_authorization_values.append({
+                    'project_summary_guid': item.get('project_summary_guid'),
+                    'project_summary_authorization_type': item.get('project_summary_authorization_type'),
+                    'authorization_type': item.get('new_type'),
+                    'authorization_description': item.get('authorization_description'),
+                    'new_type': item.get('new_type'),
+                    'exemption_requested': item.get('exemption_requested'),
+                    'is_contaminated': item.get('is_contaminated')
+                })
+
+        return new_authorization_values
+
+    @classmethod
+    def __format_phone_number(cls, phone_no):
+        return phone_no.replace('-', '').strip()
+
+    @classmethod
+    def __boolean_to_yes_no(cls, value):
+        if isinstance(value, bool):
+            return 'Yes' if value else 'No'
+        return 'No'
+
+    @classmethod
+    def create_new_ams_authorization(cls,
+                                     ams_authorizations,
+                                     applicant,
+                                     nearest_municipality,
+                                     agent,
+                                     contacts,
+                                     facility_type,
+                                     facility_desc,
+                                     facility_latitude,
+                                     facility_longitude,
+                                     facility_coords_source,
+                                     facility_coords_source_desc,
+                                     legal_land_desc,
+                                     facility_operator,
+                                     legal_land_owner_name,
+                                     legal_land_owner_contact_number,
+                                     legal_land_owner_email_address,
+                                     is_legal_land_owner,
+                                     is_crown_land_federal_or_provincial,
+                                     is_landowner_aware_of_discharge_application,
+                                     has_landowner_received_copy_of_application
+                                     ):
+        """Creates a new AMS authorization application"""
+
+        ams_results = []
+        headers = {
+            'bearer': Config.AMS_BEARER_TOKEN,
```

Comment:
Just wondering what happens if the bearer token is expired, if it can catch and refresh, or how the ams_result will be saved if it comes back as an error (will it be populated with None? trigger a 500?).